README.adoc

```diff
@@ -2185,6 +2185,28 @@ File.open('testfile') do |f|
 end
 ----
 
+=== Atomic File Operations [[atomic-file-operations]]
+
+When doing file operations after confirming the existence check of a file, frequent parallel file operations may cause problems that are difficult to reproduce.
+Therefore, it is preferable to use atomic file operations.
+
+[source,ruby]
+----
+# bad
+unless File.exist?(path)
+  FileUtils.makedirs(path)
```

Comment:
https://ruby-doc.org/stdlib-2.7.0/libdoc/fileutils/rdoc/FileUtils.html#method-c-makedirs

> `makedirs`. Alias for: `mkdir_p`.

Why is this a bad example? Would it fail if in between `exists?` and `makedirs` some other process would create the path?

Reply:
If the directory existed at the time of `exist?`, but another process erased it immediately afterwards, subsequent processes will be executed without the directory that should have been there.
However, the `makedirs` case is a case that covers a very limited problem, so `Dir.mkdir` may be a more obvious example.

An exception is raised if the directory did not exist at the time of `exist?`, but other process created it before `mkdir` was executed.